packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -375,6 +379,57 @@ export class ConstructNode {
     return false;
   }
 
+  /**
+   * Resolve a tokenized value in the context of the current Construct
+   */
+  public resolve(obj: any): any {
+    return resolve(obj, {
+      construct: this.host,
+      prefix: []
+    });
+  }
+
+  /**
+   * Record a reference originating from this construct node
+   */
+  public recordReference(ref: Token) {
+    if (ref.referenceType !== undefined && ref.referenceType !== '') {
+      this.references.add(ref);
+    }
+  }
+
+  /**
+   * Return all references of the given type originating from this node or any of its children
+   */
+  public findReferences(type: string): Token[] {
```

Comment:
`type` feels unnecessary (caller can filter the result), but I don't feel strongly about this (and it seems like you do)

Reply:
If we move `allConstructs` to here, then this becomes redundant in a sense:

```ts
const cfnRefs = construct.node.findAll().map(x => x.references).filter(x = CfnReference.isCfnReference(s));
```

Reply:
Not so succinctly, it needs to be a `flatMap` instead of a `map`, and also this does nothing to get rid of duplicates.

I guess it doesn't matter today since we're only going to have one type of Token anyway (or close enough).